nominal/core/client.py

```diff
@@ -796,23 +796,32 @@ def create_streaming_connection(
         datasource_id: str,
         connection_name: str,
         datasource_description: str | None = None,
+        nominal_data_source_rid: str | None = None,
```

Comment:
datasource vs data_source - let's figure out what the correct casing is and be consistent